src/gtfs/mod.rs

```diff
@@ -413,20 +424,58 @@ impl Reader {
             ))
         }
     }
+    /// Imports `Collections` from the
+    /// [GTFS](https://gtfs.org/reference/static).
+    /// files in the given directory.
+    /// This method will try to detect if the input is a zipped archive or not.
+    /// If the default file type mechanism is not enough, you can use
+    /// [Reader::parse_zip] or [Reader::parse_dir].
+    pub fn parse_collections(self, path: impl AsRef<Path>) -> Result<Collections> {
+        let p = path.as_ref();
+        if p.is_file() {
+            // if it's a file, we consider it to be a zip (and an error will be returned if it is not)
+            Ok(self
+                .parse_zip_collections(p)
+                .with_context(|| format!("impossible to read zipped gtfs {:?}", p))?)
+        } else if p.is_dir() {
+            Ok(self
+                .parse_dir_collections(p)
+                .with_context(|| format!("impossible to read gtfs directory from {:?}", p))?)
+        } else {
+            Err(anyhow!(
+                "file {:?} is neither a file nor a directory, cannot read a gtfs from it",
+                p
+            ))
+        }
+    }
 
     /// Imports a `Model` from a zip file containing the
     /// [GTFS](https://gtfs.org/reference/static).
     pub fn parse_zip(self, path: impl AsRef<Path>) -> Result<Model> {
-        let reader = std::fs::File::open(path.as_ref())?;
-        let mut file_handler = read_utils::ZipHandler::new(reader, path)?;
-        read_file_handler(&mut file_handler, self.configuration)
+        let collections = self.parse_zip_collections(path)?;
+        Model::new(collections)
     }
 
     /// Imports a `Model` from the [GTFS](https://gtfs.org/reference/static)
     /// files in the `path` directory.
     pub fn parse_dir(self, path: impl AsRef<Path>) -> Result<Model> {
+        let collections = self.parse_dir_collections(path)?;
+        Model::new(collections)
+    }
+
+    /// Imports `Collections` from the [GTFS](https://gtfs.org/reference/static)
+    /// files in the `path` directory.
+    fn parse_dir_collections(self, path: impl AsRef<Path>) -> Result<Collections> {
```

Comment:
We can push the idea further and put them in `pub`

Reply:
This is not necessary because `parse_collections` is `pub`

Reply:
Yes I saw. It's in case we needed it later. and to remain homogeneous.
But yes for the moment we can stay like that.